addons/mail/static/src/model/model_manager.js

```diff
@@ -639,6 +517,186 @@ class ModelManager {
+                get: (target, k) => {
+                    if (!(fieldNames.has(k))) {
+                        // No crash, we allow these reads due to patch()
+                        // implementation details that read on `this._super` even
+                        // if not set before-hand.
+                        return target[k];
+                    }
+                    return Model.fields[k].get(target);
+                },
+                set: (target, k, newVal) => {
+                    if (fieldNames.has(k)) {
+                        throw new Error("Forbidden to write on record field without .update()!!");
+                    } else {
+                        // No crash, we allow these writes due to following concerns:
+                        // - patch() implementation details that write on `this._super`
+                        // - record listeners that need setting on this with `.bind(this)`
+                        target[k] = newVal;
+                    }
+                    return true;
+                },
+            });
+            record.env = this.env;
+            record.localId = Model._createRecordLocalId(data);
+            if (Model.get(record.localId)) {
+                throw Error(`A record already exists for model "${Model.modelName}" with localId "${record.localId}".`);
+            }
+            // Contains field values of record.
+            record.__values = {};
+            // Contains revNumber of record for checking record update in useStore.
+            record.__state = 0;
+
+            Model.__records[record.localId] = record;
+
+            // Ensure X2many relations are Set initially (other fields can stay undefined).
+            for (const field of fields) {
+                if (field.fieldType === 'relation') {
+                    if (['one2many', 'many2many'].includes(field.relationType)) {
+                        record.__values[field.fieldName] = new Set();
+                    }
+                }
+            }
+
+            record._willCreate(data);
+
+            const data2 = {};
+            for (const field of fields) {
+                if (field.fieldName in data) {
+                    data2[field.fieldName] = data[field.fieldName];
+                } else {
+                    data2[field.fieldName] = field.default;
+                }
+            }
+
+            for (const field of fields) {
+                if (field.compute || field.related) {
+                    // new record should always invoke computed fields.
+                    this._registerToComputeField(record, field);
+                }
+            }
+            // new record should always invoke _updateAfter
+            this._toUpdateAfters.set(record, record._updateBefore());
+
+            this._update(record, data2);
+
+            this._createdRecords.add(record);
+
+            this._hasAnyChangeDuringCycle = true;
+
+            records.push(record);
```

Comment:
same remark as above, no empty line around most one-line statements.

I don't mine a single empty line above a chunk of statements that represent a step of a long function. This function is long, so I wouldn't mind seeing 2 or 3 such steps. Multi-line comments like in `_processDeclaredFieldsOnModels` are good too.